CDCgov/prime-reportstream
# 1172 HL7 end to end test in CLI

This PR adds an end to end tests that generates fake HL7 batch data and sends it to the ReportStream endpoint.  The tests checks the response then accesses the database to verify the lineage.  Note this test does not compare data at this point - that will be another PR.

To test:
1. Rebuild your local bbaseline with `gradle clean package`
2. Run RS locally in your docker container.
3. Run the test with `./prime test --run=hl7ingest` and verify the test is successful

## Changes
- Added new test
- Updated HL7 schema as value_type was conflicting with another value
- Added ignore-hl7 client
- Some changes to the fake data generator to provide proper data for HL7 generation
- Changes to provide proper content type when sending data from the tests
- Added a quickpackage task in gradle for devs that skips most the long tasks including tests.  Use with caution, but useful.
- Updated some documentation to be clear on how to run RS
- Updated release build to run integration tests 
- Cleaned up some compiler warnings
- Removed warnings about blank fields.  We already get errors for required fields.

## Checklist

### Testing

### Security

### Process

## Security
*List potential security threats and mitigations in the PR*

## Fixes
- #1172

## To Be Done
Add data comparison to end to end test
- #1345
